scarb/tests/fmt.rs

```diff
@@ -270,3 +304,79 @@ fn workspace_with_root() {
     let content = fs::read_to_string(t.child("second/src/lib.cairo")).unwrap();
     assert_eq!(content, SIMPLE_FORMATTED);
 }
+
+#[test]
+fn workspace_emit_with_root() {
+    let t = TempDir::new().unwrap().child("test_workspace");
+    let pkg1 = t.child("first");
+    ProjectBuilder::start()
+        .name("first")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .build(&pkg1);
+    let pkg2 = t.child("second");
+    ProjectBuilder::start()
+        .name("second")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .dep("first", &pkg1)
+        .build(&pkg2);
+    let root = ProjectBuilder::start()
+        .name("some_root")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .dep("first", &pkg1)
+        .dep("second", &pkg2);
+    WorkspaceBuilder::start()
+        .add_member("first")
+        .add_member("second")
+        .package(root)
+        .build(&t);
+
+    Scarb::quick_snapbox()
+        .arg("fmt")
+        .arg("--emit")
+        .arg("stdout")
+        .current_dir(&t)
+        .assert()
+        .failure()
+        .stdout_matches(format!(
+            "{}:\n{}\n",
+            fs::canonicalize(t.child("src/lib.cairo"))
```

Comment:
```suggestion
            fsx::canonicalize(t.child("src/lib.cairo"))
```
Please use `fsx` instead, it's our internal extension of `fs`.